cocos/2d/renderer/batcher-2d.ts

```diff
@@ -615,7 +615,7 @@ export class Batcher2D {
 
     private _preProcess (node: Node) {
         const render = node._uiProps.uiComp;
-        if (!render) { // hack for opacity
+        if (!render || !render.isRender2D) { // hack for opacity
```

Comment:
Please find another way to implement this feature. Opacity could be stored in batcher2d and keep inheritance valid through the hierarchy